Player movement now in ECS.  Camera and map both back up and running.  E...

gcoetz opened the pull request:
So what we have is the player moving system which just looks if there’s keyboard input and moves the player accordingly.
The camera updating system looks for the player entity and updates the renderer’s camera to the right position based on boundaries and the player’s location.
The MapBoundrySystem is there just to make the player (and monsters, in theory) not walk off the map.  I think this should turn into or be replaced by a physics system that just checks for impassable tiles on the map, and we shouldn’t have to check the map bounds for entities when we’re at that point.

The part of the design I wasn’t too sure about was attaching the map to the renderer.  I’ve been trying to keep physics and rendering separate the map is needed for both.

Ps. Sorry this coming in so late… I thought I would have time to do the pull-request last night but it never happened >.<

gcoetz committed:
Player movement now in ECS.  Camera and map both back up and running.  Entities no longer wander off the map (temporary solution)

sturmE commented on DirtyGame/Game/Core/Systems/Render/CameraUpdateSystem.cs:
```diff
@@ -0,0 +1,77 @@
+
+                newPos = CheckMapBounds(newPos);
+
+                Camera newCam = new Camera();
```
You create a new camera every frame and set its position? Is that intended?

gcoetz replied:
There was no good reason for doing this, no.

sturmE commented on DirtyGame/Game/Core/Systems/PlayerControlSystem.cs:
```diff
@@ -0,0 +1,69 @@
+        private KeyboardState KeyboardState;
+
+        public PlayerControlSystem()
+            : base(SystemDescriptions.SpriteRenderSystem.Aspect, SystemDescriptions.SpriteRenderSystem.Priority)
```
You created an aspect for this system but arent using it.

sturmE commented:
Nice job. I like.

JaredPoetter commented on DirtyGame/Game/Core/Systems/PlayerControlSystem.cs:
```diff
@@ -0,0 +1,69 @@
+
+            foreach (Entity e in entities)
+            {
+                if (!e.HasComponent<Player>()) continue;
```
Like Eugene was talking about above with the Aspect. You could make this code simpler because if you put the Player component in the Aspect for the PlayerControlSystem it will only run this system on entities that have the Player component. So this should improve the performance of this system a lot because right now you are needing to loop through all of the entities to find the one, in this case, to move with the input.

gcoetz replied:
Yup yup you guys were right, updated with the right system description name.

gcoetz committed:
Used the right system description for PlayerControlSystem...
Changed camera update to no longer make a new camera every time (small optimization)

mplewis commented:
Can we rename `boundry` to `boundary` before this gets merged?

mplewis commented on DirtyGame/Game/Core/Systems/Util/SystemDescriptions.cs:
```diff
@@ -12,6 +12,16 @@ public class SystemDescriptions
         // This needs to be last
         public static SystemParams SpriteRenderSystem =
-            new SystemParams(Aspect.CreateAspectFor(new List<Type> {typeof (Sprite), typeof (Spatial)}), 1000);
+            //new SystemParams(Aspect.CreateAspectFor(new List<Type> {typeof (Sprite), typeof (Spatial)}), 1000);
```
Please remove this commented line.

mplewis commented on DirtyGame/DirtyGame.csproj.user:
```diff
@@ -0,0 +1,13 @@
+    <FallbackCulture>en-US</FallbackCulture>
+    <VerifyUploadedFiles>false</VerifyUploadedFiles>
+  </PropertyGroup>
+</Project>
```
Please add a newline at the end of this file.

sturmE replied:
This is csproj file, its autogenerated.

mplewis replied:
Ah ok. Retracted! :sushi:

sturmE commented:
Dont know why github didnt close this pull request. It has been merged.
